rotkehlchen/db/history_events.py

```diff
@@ -226,18 +257,27 @@ def get_history_events(
         Get history events using the provided query filter
         """
         query, bindings = filter_query.prepare()
-
-        if has_premium:
-            query = 'SELECT * from history_events ' + query
-            cursor.execute(query, bindings)
+        if isinstance(filter_query, EvmEventFilterQuery):
+            # Select only evm events if `EvmEventFilterQuery` is used
+            join_query = EVM_EVENT_JOIN
         else:
-            query = 'SELECT * FROM (SELECT * from history_events ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) ' + query  # noqa: E501
-            cursor.execute(query, [FREE_HISTORY_EVENTS_LIMIT] + bindings)
+            join_query = ALL_EVENTS_DATA_JOIN
+
+        base_query = f'SELECT {HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS} {join_query}'
+        if has_premium is False:
+            base_query = f'SELECT * FROM ({base_query} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '  # noqa: E501
+            bindings.insert(0, FREE_HISTORY_EVENTS_LIMIT)
```

Comment:
no need to assign to `base_query` twice when `has_premium` is False

```suggestion
        if has_premium is True:
            base_query = f'SELECT {HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS} {join_query}'
        else:
            base_query = f'SELECT * FROM ({base_query} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '  # noqa: E501
            bindings.insert(0, FREE_HISTORY_EVENTS_LIMIT)
```